SQL/Release_patches/22.0_To_23.0_upgrade.sql

```diff
@@ -0,0 +1,480 @@
+-- -----------------------------------------------------------------------------------
+ALTER TABLE `mri_violations_log` ADD COLUMN `MriProtocolChecksGroupID` INT(4) UNSIGNED NOT NULL;
+UPDATE mri_violations_log SET MriProtocolChecksGroupID=(SELECT MriProtocolChecksGroupID FROM mri_protocol_checks_group WHERE Name='Default MRI protocol checks group');
+ALTER TABLE `mri_violations_log` ADD CONSTRAINT `FK_mri_checks_group_1` FOREIGN KEY (`MriProtocolChecksGroupID`) REFERENCES `mri_protocol_checks_group` (`MriProtocolChecksGroupID`);
+
+
+SELECT 'Running: SQL/New_patches/2019-07-02-Add_Edit_DoB_Permissions.sql';
+
+INSERT INTO permissions (code, description, categoryID) VALUES
+    ("candidate_dob_edit","Edit dates of birth",2);
+    
+INSERT INTO user_perm_rel VALUES
+    (1, (SELECT permID FROM permissions WHERE code='candidate_dob_edit'));
+
+SELECT 'Running: SQL/New_patches/2019-07-04-add_DoD_feature.sql';
+
+ALTER TABLE candidate
+    ADD COLUMN DoD date default NULL;
+
+ -- candidate_dod_edit permission
+INSERT IGNORE INTO permissions (code, description, categoryID) VALUES
+    ('candidate_dod_edit', "Edit dates of death", 2);
+
+INSERT IGNORE INTO user_perm_rel (userID, permID) VALUES
+    (1, (SELECT permID FROM permissions where code='candidate_dod_edit'));
+
+INSERT INTO ConfigSettings (Name, Description, Visible, AllowMultiple, DataType, Parent, Label, OrderNumber)
+    SELECT 'dodFormat', 'Format of the Date of Death', 1, 0, 'text', ID, 'DOD Format', 10 FROM ConfigSettings WHERE Name="study";
+
+INSERT INTO Config (ConfigID, Value) SELECT ID, 'YMd'  FROM ConfigSettings WHERE Name="dodFormat";
+
+SELECT 'Running: SQL/New_patches/2019-07-05-Add_Language_feature.sql';
+
+ALTER TABLE `session`
+    ADD COLUMN `languageID` integer unsigned DEFAULT NULL,
+    ADD CONSTRAINT `FK_session_4` FOREIGN KEY (`languageID`) REFERENCES `language` (`language_id`);
+
+SELECT 'Running: SQL/New_patches/2019-08-06-Add_date_display_format_config_setting.sql';
+
+INSERT INTO ConfigSettings (Name, Description, Visible, AllowMultiple, DataType, Parent, Label, OrderNumber) SELECT 'dateDisplayFormat', 'The date format to use throughout LORIS for displaying date information - formats for date inputs are browser- and locale-dependent.', 1, 0, 'text', ID, 'Date display format', 27 FROM ConfigSettings WHERE Name="study";
+INSERT INTO Config (ConfigID, Value) SELECT ID, 'Y-m-d H:i:s'  FROM ConfigSettings WHERE Name="dateDisplayFormat";
+
+SELECT 'Running: SQL/New_patches/2019-10-09-move_MINCToolsPath_configuration_to_Config_tables.sql';
+
+INSERT INTO ConfigSettings (Name, Description, Visible, AllowMultiple, DataType, Parent, Label, OrderNumber) SELECT 'MINCToolsPath', 'Path to the MINC tools', 1, 0, 'web_path', ID, 'Path to the MINC tools', 12 FROM ConfigSettings WHERE Name="paths";
+INSERT INTO Config (ConfigID, Value) SELECT ID, "%MINCToolsPath%" FROM ConfigSettings WHERE Name="MINCToolsPath";
```

Comment:
It is better to use this:
INSERT INTO Config (ConfigID, Value) SELECT ID, "%MINCToolsPath%" FROM ConfigSettings WHERE Name="MINCToolsPath"  and NOT EXISTS (Select * from Config
where ConfigID=(SELECT ID FROM ConfigSettings WHERE Name="MINCToolsPath"));
otherwise run this patch twice accidentlly, Loris will be broken.